src/server/rpc/rpc-manager.js

```diff
@@ -153,10 +152,16 @@ RPCManager.prototype.handleRPCRequest = function(RPC, req, res) {
         }
 
         // Add the netsblox socket for triggering network messages from an RPC
-        rpc.socket = SocketManager.getSocket(uuid);
-        rpc.response = res;
-        if (!rpc.socket) {
-            this._logger.error(`Could not find socket ${uuid} for rpc ` + 
+        let ctx;
+        if (rpc instanceof ApiConsumer) {
```

Comment:
For now, this is fine since I know it is urgent but we should remove this and apply this technique for all the rpcs (storing the meaningful info in `this._state`)